lib/install.js

```diff
@@ -26,11 +26,12 @@ async function install(parentDir, pkg, ancestors, options, context) {
     return await _install(parentDir, pkg, ancestors, options, context);
   } catch (err) {
     if (pkg.optional) {
-      let message = err.stack;
       if (err.name === 'UnSupportedPlatformError') {
-        message = err.message;
+        // ignore log error
+        debug(err);
+        return;
```

Comment:
Nice work. This is consistent with npm7, the warning message can be scary.

https://github.com/npm/npm/issues/9567